### ISSUE TITLE
Docker build errors due to missing npm

### ISSUE THREAD

**Issue by jornh** [lyft/amundsenfrontendlibrary#117]
When doing a `docker build ....` in this repo you’ll experience the error message shared over in the metadata repo in this issue thread: https://github.com/lyft/amundsenmetadatalibrary/issues/20#issuecomment-488056349 (see container build output shared by @rosejcday further up in the issue over there.

I suspect the reason for the failure is that the python docker base image doesn’t contain node and npm.

A good solution to fix this would be to use docker multi-stage build. With that you use one `node` development container to build the react/Javascript “webpacked” `dist/` output and copy that over to the final python based container. Advantages of this approach is smaller final runtime container (no need for node at runtime) - potentially less moving parts/vulnerability surface in the final runtime container.

**Comment by gschmutz** [comment]
I'm having a problem with the docker image as well. When I start the environment using the docker-compse.

`
amundsenfrontend    | 2019-05-01T17:57:21+0000.082 [ERROR] app.log_exception:1761 (1:Thread-3) - Exception on /favicon.ico [GET]
amundsenfrontend    | Traceback (most recent call last):
amundsenfrontend    |   File "/usr/local/lib/python3.7/site-packages/flask/app.py", line 2292, in wsgi_app
amundsenfrontend    |     response = self.full_dispatch_request()
amundsenfrontend    |   File "/usr/local/lib/python3.7/site-packages/flask/app.py", line 1815, in full_dispatch_request
amundsenfrontend    |     rv = self.handle_user_exception(e)
amundsenfrontend    |   File "/usr/local/lib/python3.7/site-packages/flask/app.py", line 1718, in handle_user_exception
amundsenfrontend    |     reraise(exc_type, exc_value, tb)
amundsenfrontend    |   File "/usr/local/lib/python3.7/site-packages/flask/_compat.py", line 35, in reraise
amundsenfrontend    |     raise value
amundsenfrontend    |   File "/usr/local/lib/python3.7/site-packages/flask/app.py", line 1813, in full_dispatch_request
amundsenfrontend    |     rv = self.dispatch_request()
amundsenfrontend    |   File "/usr/local/lib/python3.7/site-packages/flask/app.py", line 1799, in dispatch_request
amundsenfrontend    |     return self.view_functions[rule.endpoint](**req.view_args)
amundsenfrontend    |   File "/usr/local/lib/python3.7/site-packages/amundsen_frontend-1.0.2-py3.7.egg/amundsen_application/api/__init__.py", line 13, in index
amundsenfrontend    |     return render_template("index.html")  # pragma: no cover
amundsenfrontend    |   File "/usr/local/lib/python3.7/site-packages/flask/templating.py", line 134, in render_template
amundsenfrontend    |     return _render(ctx.app.jinja_env.get_or_select_template(template_name_or_list),
amundsenfrontend    |   File "/usr/local/lib/python3.7/site-packages/jinja2/environment.py", line 869, in get_or_select_template
amundsenfrontend    |     return self.get_template(template_name_or_list, parent, globals)
amundsenfrontend    |   File "/usr/local/lib/python3.7/site-packages/jinja2/environment.py", line 830, in get_template
amundsenfrontend    |     return self._load_template(name, self.make_globals(globals))
amundsenfrontend    |   File "/usr/local/lib/python3.7/site-packages/jinja2/environment.py", line 804, in _load_template
amundsenfrontend    |     template = self.loader.load(self, name, globals)
amundsenfrontend    |   File "/usr/local/lib/python3.7/site-packages/jinja2/loaders.py", line 113, in load
amundsenfrontend    |     source, filename, uptodate = self.get_source(environment, name)
amundsenfrontend    |   File "/usr/local/lib/python3.7/site-packages/flask/templating.py", line 58, in get_source
amundsenfrontend    |     return self._get_source_fast(environment, template)
amundsenfrontend    |   File "/usr/local/lib/python3.7/site-packages/flask/templating.py", line 86, in _get_source_fast
amundsenfrontend    |     raise TemplateNotFound(template)
amundsenfrontend    | jinja2.exceptions.TemplateNotFound: index.html
`

I have added the following #121 PR which solves the issue with a multi-stage docker build, as suggested by @jornh.

**Comment by jornh** [comment]
I did a docker pull of the *new version from today* of 1.0.5 (actually by setting the `:1.05` instead of `:latest` tag in `docker.amundsen.yml`). 

Seems to work fine now, and includes `index.html` and other FE payload as expected.

I propose we start tagging versions for docker images in the compose files. This ensures people get well know and reproducible combinations of containers (just like versions are also pegged in Python `requirements.txt` and JS `project.json` files).

I will push a PR with this for what I have running locally in a bit - though merge of that is awaiting that we have a new tagged `1.0.2` for the search service (so we get the Neo4j fix I saw just got merged to `master` over there) 

Closing this issue as resolved by #121 - thanks @gschmutz @jinhyukchang - CC @feng-tao
